dali/operators/reader/numpy_reader_op.cc

```diff
@@ -116,4 +113,52 @@ DALI_SCHEMA(NumpyReader)
 submodule and renamed to follow a common pattern. This is a placeholder operator with identical
 functionality to allow for backward compatibility.)code");  // Deprecated in 1.0;
 
+bool NumpyReader::SetupImpl(std::vector<OutputDesc> &output_desc,
+                            const workspace_t<CPUBackend> &ws) {
+  // If necessary start prefetching thread and wait for a consumable batch
+  DataReader<CPUBackend, ImageFileWrapper>::SetupImpl(output_desc, ws);
+
+  const auto &file_0 = GetSample(0);
+  TypeInfo output_type = file_0.image.type();
+  int ndim = file_0.image.shape().sample_dim();
+
+  TensorListShape<> sh(max_batch_size_, ndim);
+  for (int i = 0; i < max_batch_size_; i++) {
```

Comment:
Maybe instead of `max_batch_size_` it should check size of `prefetched_batch_queue_[curr_batch_consumer_]`. Just for the future.